src/components/PopupWithConfirmation.js

```diff
@@ -7,14 +8,27 @@ export default class PopupWithConfirmation extends Popup {
   }
 
   setSubmitAction(submitAction) {
-    this._confirmButton.addEventListener("mousedown", () => {
-      return submitAction(this._card);
-    });
+    this._submitAction = submitAction;
   }
 
-  open(card) {
+  open(card, cardId) {
     super.open();
     this._card = card;
+    this._cardId = cardId;
   }
 
+  setEventListeners() {
+    super.setEventListeners();
+
+    this.setSubmitAction(() => {
+      api.deleteCard(this._cardId)
+        .then(() => {
+          this._card.removeCard();
+          this.close();
+        })
+        .catch(catchError)
+    });
```

Comment:
You don't need `setSubmitAction` anymore because you update the card and its id.

The code will be simplier here


```js
  setEventListeners() {
    super.setEventListeners();

    this._confirmButton.addEventListener("click", () => {
      this._handleSubmit(this._cardId, this._card )
    });
  }

```